### ISSUE TITLE
[Cognitive Services - Anomaly Detector] Multidimensional Root Cause Analysis

### ISSUE THREAD

**Issue by suxi-ms** [Azure/azure-rest-api-specs#16984]
• Key contact for the review:  suxi@microsoft.com
• Whether this is a new or existing API:  This is a set of new API
• Whether or not your service already been deployed:   Hasn't been deployed
• A brief description of the material to be reviewed/change:  This new feature from Anomaly Detector service is Multidimensional Root Cause Analysis (MDRCA). It can discover the root causes of problems to identify appropriate solutions within your complex metric systems such as ROI, CTR, through detecting the abnormal dimension(s) or abnormal dimension combinations when there is a big spike or drop in your metrics system, without knowing any machine learning knowledge, labeled data, or even anomaly information.
• A link to the PR/issue in  azure/azure-rest-api-specs [link](https://github.com/Azure/azure-rest-api-specs/pull/18037)

**Comment by czubair** [comment]
Please review updated PR: https://github.com/Azure/azure-rest-api-specs/pull/17083

**Comment by czubair** [comment]
2nd review scheduled for 1/13 at 4pm PST

**Comment by markweitzel** [comment]
## API Stewardship Review 13-Jan-22

### Discussion / Notes
* `PartitionIdentity` Consider the name--an identity should not be optional, so think of another name. 
* Should have better description of the data. 
* `PUT` should end in a path parameter, e.g. `instanceId`
  * If there is second `PUT` and the data is the same, this is successful. If the data is not the same, then this should be a `4xx` error (right now `409` but let's confirm off-line @JeffreyRichter )
* `POST` for `.../inferences` should be a `PUT` with a modelId on the end.

Where possible, use the `PUT` / `GET` pattern. A proposed approach, to keep the model of timestamps having partitions.
* `PUT` `/rca/datsaSets/{dataSetId}/timestamps/{timestamp}/partitions/{partitionId}`
* `DELETE` `/rca/datsaSets/{dataSetId}/timestamps/{timestamp}/partitions/{partitionId}`
* `POST` `/rca/dataSets/{dataSetId}/timestamps:upload`

`DELETE` `/rootCauseAnalysis/dataSets/{dataSetId}/data`

**Comment by JeffreyRichter** [comment]
Yes, 409 seems right to me

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by czubair** [comment]
Next review scheduled for 3/3 at 4pm PST. Updated PR: https://github.com/Azure/azure-rest-api-specs/pull/17083

**Comment by suxi-ms** [comment]
New PR link：https://github.com/Azure/azure-rest-api-specs/pull/18037

**Comment by markweitzel** [comment]
## API Stewardship Board Review Meeting 3-March-2022


### Discussion / Notes
* Make sure that `dataSize` will not exceed -2^53+1 to +2^53-1 
* Consider changing the name of the `dataSize` to `count` (or something similar, e.g. `rowCount`)
* Note: It is only possible to upload and delete the data, you cannot download it.

* The filter expression language should more closely align to OData specification. 
  * See [OData](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html) 
  * https://docs.microsoft.com/en-us/azure/search/search-query-odata-filter
  * https://azuresearch.github.io/odata-syntax-diagram

### Actions
@achandmsft & @robch -- Is there a standard filter format that we are using across Cognitive Services?


## References
[Meeting Chat]()
[Meeting Recording]()

@suxi-ms -- [ADD LINK TO HERO SCENARIOS HERE]

### Helpful Documentation
* [Azure API REST Documentation](https://aka.ms/azapi/specs)
* [OData](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html) 
* https://docs.microsoft.com/en-us/azure/search/search-query-odata-filter
* https://azuresearch.github.io/odata-syntax-diagram

**Comment by JeffreyRichter** [comment]
I watched this meeting's recording about the filter discussion. Yes, the filter should be odata-ish (which was not what was shown) but most services would support a severe subset of all the odata functionality. 

A few weeks ago, I decided (on a whim) to build a lexer/scanner/evaluator in Go to parse & evaluate filter strings just to see how difficult it would be for a service team to implement filtering. My code is about 1000 lines of Go which I offer [here](https://github.com/JeffreyRichter/Filter) if the team wants to use it. Since I wrote it in Go, it would have to be converted to whatever language the team is using. In the video, I saw "contains(string, value)" and my code does not currently support this function; it would have to be added. I might have time in the next week or so to add support for this - it shouldn't be too difficult.
